### PR TITLE
AutoConfiguration support for Data Mongo ReactiveGridFsTemplate

### PR DESCRIPTION
**Pull request by ayudovin**
this [enhancement](https://github.com/spring-projects/spring-boot/issues/16393)

### CODE REVIEW

**Comment by ayudovin** [comment]
Test failure seems unrelated

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/mongo/MongoReactiveDataAutoConfiguration.java [hunk 1]
```diff
@@ -85,4 +87,12 @@ public MappingMongoConverter mappingMongoConverter(MongoMappingContext context,
 		return mappingConverter;
 	}
 
+	@Bean
+	@ConditionalOnMissingBean
+	public ReactiveGridFsTemplate reactiveGridFsTemplate(
+			ReactiveMongoDatabaseFactory reactiveMongoDbFactory,
+			MappingMongoConverter mappingMongoConverter) {
+		return new ReactiveGridFsTemplate(reactiveMongoDbFactory, mappingMongoConverter);
```
Comment (mp911de) [inline spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/mongo/MongoReactiveDataAutoConfiguration.java]: Ideally, we should configure `ReactiveGridFsTemplate ` with `DataBufferFactory` to share potentially resources.

Comment (ayudovin) [reply spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/mongo/MongoReactiveDataAutoConfiguration.java]: `ReactiveGridFsTemplate` constructor has 4 parameters:

- `DataBufferFactory`

- `ReactiveMongoDatabaseFactory`

- `MappingMongoConverter`

- `bucket`

What value of bucket should be in this case?

Comment (mp911de) [reply spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/mongo/MongoReactiveDataAutoConfiguration.java]: Defaults to `null`, see https://github.com/spring-projects/spring-data-mongodb/blob/master/spring-data-mongodb/src/main/java/org/springframework/data/mongodb/gridfs/ReactiveGridFsTemplate.java#L68.

File: spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/data/mongo/MongoReactiveDataAutoConfigurationTests.java [hunk 2]
```diff
@@ -55,4 +56,10 @@ public void backsOffIfMongoClientBeanIsNotPresent() {
 				.doesNotHaveBean(MongoReactiveDataAutoConfiguration.class));
 	}
 
+	@Test
+	public void gridFsTemplateExists() {
+		this.contextRunner.run((context) -> assertThat(context)
+				.hasSingleBean(ReactiveMongoTemplate.class));
```
Comment (snicoll) [inline spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/data/mongo/MongoReactiveDataAutoConfigurationTests.java]: This looks like you're checking for the wrong bean there.

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/mongo/MongoReactiveDataAutoConfiguration.java [hunk 3]
```diff
@@ -85,4 +89,20 @@ public MappingMongoConverter mappingMongoConverter(MongoMappingContext context,
...
+			ReactiveMongoDatabaseFactory reactiveMongoDbFactory,
+			MappingMongoConverter mappingMongoConverter,
+			DataBufferFactory dataBufferFactory) {
+		return new ReactiveGridFsTemplate(dataBufferFactory, reactiveMongoDbFactory,
+				mappingMongoConverter, null);
+	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	public DataBufferFactory dataBufferFactory() {
```
Comment (snicoll) [inline spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/mongo/MongoReactiveDataAutoConfiguration.java]: Can we flip those two methods? Required dependencies are defined before so I'd like to keep it that way.

**Comment by snicoll** [comment]
@ayudovin thanks for the feedback

**Pull request closed without merging**